### DISCUSSION TITLE
-no-user-cache option doesn't seem to work anymore

### DISCUSSION THREAD

**Discussion by pican79** [rusq/slackdump#77] [Q&A] [answered]
With the [v2.0.0-rc.2 release](https://github.com/rusq/slackdump/releases/tag/v2.0.0-rc.2), I always get this error when I specify the `-no-user-cache` option:
`invalid number of users retrieved`

In an [older version of the readme](https://github.com/rusq/slackdump/tree/0b629a5a97f58001bfa886e968915874c60390b1), you were wondering why anyone would use this flag:

> -no-user-cache
    skip fetching users. If this flag is specified, users won't be fetched during startup. This disables the username resolving for the text output, I don't know why someone would use this flag, but it's there if you must.

The reason I used it is because my workspace has several thousands users and it takes a long time to fetch them (because of my slow internet connection?).

The channels/threads I use slackdump on have very few users and they rarely change.

One thing I noticed with the v2.0.0-rc.2 release is that the users cache file is now stored in `%LOCALAPPDATA%\slackdump`.
With previous releases, it was in whatever folder I ran the slackdump executable from.

**Comment by rusq** [comment]
Hey @pican79 , thanks for the discovery!

Found the reason - this regression is related to the Slack Export imlementation (in `slackdump.go`):
```diff
+       if len(users) < 2 { // me and slackbot.
+               return nil, errors.New("invalid number of users retrieved.")
+       }
+
+       // now, this is filthy, but Slack does not allow us to call GetUserIdentity with browser token.
+       sd.me = users[userIdxMe]
```

For the slack export we need to know the ID of the user that is running the export, and this user always has index 1 in the list of retrieved users.  It is not possible to use GetUserIdentity call, because it requires different kind of token, so I resorted to a filthy way of getting the "current user", and paid the price of breaking --no-user-cache function.  Will fix.

**Reply by rusq** [reply]
You are correct in your observation of the location for the cache - I now use the OS cache directory to save the user cache in order to keep the program's folder tidy.

Also, check out new -base parameter, which makes slackdump store jsons not in the current directory, but in some other, i.e. `-base "blah"`.  It also supports saving data into a ZIP file, i.e. `-base "blah.zip"`

**Comment by rusq** [comment, marked as answer]
Fixed in v2 release :)
